aad/client.bal

```diff
@@ -0,0 +1,423 @@
+    # 
+    # +'type - The type of directiry object
+    # + objectId - The object ID
+    # + queryParams - Optional query parameters. This method support OData query parameters to customize the response.
+    #                 It should be an array of type `string` in the format `<QUERY_PARAMETER_NAME>=<PARAMETER_VALUE>`
+    #                 **Note:** For more information about query parameters, refer here: 
+    #                   https://docs.microsoft.com/en-us/graph/query-parameters
+    # + return - A stream of type `Group` if sucess. Else `error`.
+    @display {label: "List Parent Groups"}
+    remote isolated function listParentGroups(@display {label: "Directory Object Type"} DirctoryObject 'type, 
+                                              @display {label: "Object ID"} string objectId, 
+                                              @display {label: "Optional Query Parameters"} string[] queryParams = []) 
+                                              returns stream<Group, error?>|error {
+        string path = EMPTY_STRING;
+        if('type == USER) {
+            path = check createUrl([USERS, objectId, MEMBER_OF], queryParams);   
+        } else {
+            path = check createUrl([GROUPS, objectId, MEMBER_OF], queryParams);   
+        }
+        
+        http:Response response = check self.httpClient->get(path);
+        map<json>|string handledResponse = check handleResponse(response);
+        GroupStream objectInstance = check new (self.httpClient, path);
+        stream<Group, error?> finalStream = new (objectInstance);
+        return finalStream; 
+    }
+
+    # Get groups that the group is a member of. This operation is transitive and will also include all groups that this 
+    # groups is a nested member of.
+    # 
+    # +'type - The type of directiry object
+    # + objectId - The object ID
+    # + queryParams - Optional query parameters. This method support OData query parameters to customize the response.
+    #                 It should be an array of type `string` in the format `<QUERY_PARAMETER_NAME>=<PARAMETER_VALUE>`
+    #                 **Note:** For more information about query parameters, refer here: 
+    #                   https://docs.microsoft.com/en-us/graph/query-parameters
+    # + return - A stream of type `Group` if sucess. Else `error`.
+    @display {label: "List Transitive Parent Groups"}
+    remote isolated function listTransitiveParentGroups(@display {label: "Directory Object Type"} DirctoryObject 'type, 
+                                                        @display {label: "Object ID"} string objectId, 
+                                                        @display {label: "Optional Query Parameters"} 
+                                                        string[] queryParams = []) returns 
+                                                        stream<Group, error?>|error {
+        string path = EMPTY_STRING;
+        if('type == "user") {
```

Comment:
In some places USER is used. Shall we use that here?

Reply:
Fixed thanks